### PR TITLE
Fix generation of purecap kernel MFS root kernel config.

### PR DESCRIPTION
**Pull request by qwattash**
Prevent appending duplicate -NODEBUG.

### CODE REVIEW

**Comment by jrtc27** [comment]
Why? I deliberately fixed it so -NODEBUG was preserved.

**Comment by jrtc27** [comment]
Ok, the problem is BuildCheriBsdMfsKernel.process unconditionally adds `benchmark_suffix`. It should only do so if not already present.

**Comment by qwattash** [comment]
@jrtc27 Latest purecap jenkins build fails because there is a duplicate -NODEBUG appended. 
https://ctsrd-build.cl.cam.ac.uk/blue/organizations/jenkins/CheriBSD-pipeline/detail/PR-614/22/pipeline/790/
I think this should generate the correct set of kernels, I tried locally a pretend build.

**Comment by qwattash** [comment]
Let me change it so that it appends conditionally, probably more robust

**Comment by jrtc27** [comment]
Maybe we want it to instead _strip_ it in that case, if the objective is to build all the variants?

**Comment by qwattash** [comment]
Yes, I think the current change in this PR should stay there, as the default kernel should be the neutral variant, and then we append suffixes to it, if needed.

**Comment by jrtc27** [comment]
No I meant that `BuildCheriBsdMfsKernel.process` should be:

```python3
        # also build the benchmark/normal kernel:
        if benchmark_suffix:
            if default_kernconf.endswith(benchmark_suffix):
                kernel_configs.append(default_kernconf[0:-len(benchmark_suffix)])
            else:
                kernel_configs.append(default_kernconf + benchmark_suffix)
```

**Comment by jrtc27** [comment]
And similarly for fpga_conf later

**Comment by jrtc27** [comment]
Your solution breaks if `benchmark_suffix` is `None`, no?

**Comment by jrtc27** [comment]
Also, `strip`'s string argument is a set of characters to strip, so it will remove anything matching `[-NODEBUG]`, which is not what you want.

**Comment by qwattash** [comment]
`strip(None)` behaves as `strip()` with no argument, unless some older python version does something weird. So it should be fine, if a bit contrived.

**Comment by jrtc27** [comment]
.. which then strips whitespace?

**Comment by qwattash** [comment]
There should be no whitespace at the beginning or end of the string for a file name, I would hope. But I can go with the more explicit solution if it reads better.

**Comment by jrtc27** [comment]
My point is, `strip` is entirely the wrong interface. It's broken if you have a non-None `benchmark_suffix` due to being a character set not a suffix (and even if it were what you think it is it'd match *prefixes* too), and if it's None then it tries to do a bunch of work that ends up being a no-op. Python 3.9 finally adds nice methods to remove prefixes and suffixes, but that's very new.

**Comment by qwattash** [comment]
Ok, I agree that I was doing something silly. Should be correct now.

File: pycheribuild/projects/cross/cheribsd.py [hunk 1]
```diff
@@ -319,7 +319,7 @@ def default_kernel_config(self):
                         return "CHERI-CAPREVOKE-QEMU-FETT"
                     return "CHERI-CAPREVOKE-QEMU"
                 if self.purecap_kernel:
-                    return "CHERI-PURECAP-QEMU-NODEBUG"
+                    return "CHERI-PURECAP-QEMU"
```
Comment (jrtc27) [inline pycheribuild/projects/cross/cheribsd.py:322]: Please revert this; we don't want everyone to be waiting twice as long for QEMU to boot...

**Comment by jrtc27** [comment]
Looks fine now from my perspective but please update the commit message to reflect the new approach.

**Review by arichardson — APPROVED** [review]

**Comment by bsdjhb** [comment]
Wasn't the slow boot with DEBUG made better when we added WITNESS_SKIPSPIN to riscv's GENERIC kernel?

**Comment by jrtc27** [comment]
I don't know if anyone measured the difference for the purecap kernel.

**Comment by arichardson** [comment]
> Wasn't the slow boot with DEBUG made better when we added WITNESS_SKIPSPIN to riscv's GENERIC kernel?

It "feels" faster after WITNESS_SKIPSPIN but I didn't do any measurements.

**Pull request merged**